core/src/main/scala/chisel3/Module.scala

```diff
@@ -117,6 +118,14 @@ object Module extends SourceInfoDoc {
     *   direction given by upper-levels (override Input / Output)
     */
   private[chisel3] def assignCompatDir(data: Data): Unit = {
+    // Collect all leaf elements of the data which have an unspecified or flipped
+    // direction, and assign explicit directions to them
+    DataMirror
+      .collectMembers(data) {
+        case x: Element if (x.specifiedDirection == SpecifiedDirection.Unspecified | SpecifiedDirection.Flip) => x
```

Comment:
this is legal scala? I thought you'd have to write (x.specifiedDirection == a || x.specifiedDirection == B)

Reply:
You're right, I just made a force push to correct this